src/control/cmd/daos_server/support.go

```diff
@@ -0,0 +1,94 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/daos-stack/daos/src/control/common/cmdutil"
+	"github.com/daos-stack/daos/src/control/lib/support"
+)
+
+type SupportCmd struct {
+	CollectLog collectLogCmd `command:"collectlog" description:"Collect logs from server"`
+}
+
+// collectLogCmd is the struct representing the command to collect the Logs/config for support purpose
+type collectLogCmd struct {
+	cfgCmd
+	cmdutil.LogCmd
+	Stop         bool   `short:"s" long:"Stop" description:"Stop the collectlog command on very first error"`
+	TargetFolder string `short:"t" long:"loglocation" description:"Folder location where log is going to be copied"`
+	Archive      bool   `short:"z" long:"archive" description:"Archive the log/config files"`
+	CustomLogs   string `short:"c" long:"custom-logs" description:"Collect the Logs from given directory"`
+}
+
+func (cmd *collectLogCmd) Execute(_ []string) error {
```

Comment:
So much of this function seems identical to the one in the agent. Could most of this be refactored into a CollectAllLogs function in the new `lib/support` package you created? Seems like it takes the params above, plus the LogCollection steps.